O-Osahon/Epic-Mail
# #164739805 Postgresql Test Environement

### What does this PR do?
Integrates a Postgresql database for testing.
Refactors codebase; uses `async - await` rather than the previously used `promises`

### Description of tasks to be completed
- create a test database instance on a local machine
- conditionally connect the database to the application (only when `NODE_ENV` is set to `test`)
- seed mock data to `messages` and `users` tables

### How this can be manually tested
- in the terminal run `git clone https://github.com/O-Osahon/Epic-Mail.git.`
- run `git checkout ch-db-test-env-164739805`
- run `npm run prep-test`
- run `npm run db:populate`
after this, the database should be populated with the data in the `/server/models/mock`

### What are the relevant Pivotal Tracker stories?
[#164739805](https://www.pivotaltracker.com/story/show/164739805)
